pkg/provider/aws/static_unit.go

```diff
@@ -4,20 +4,47 @@ import (
 	"path/filepath"
 
 	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/multierr"
 	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
 )
 
-func (a *AWS) GenerateStaticUnitResources(unit *core.StaticUnit, dag *core.ResourceGraph) error {
+//func (a *AWS) GenerateStaticUnitResources(unit *core.StaticUnit, dag *core.ResourceGraph) error {
+//
+//	bucket := resources.NewS3Bucket(unit, a.Config.AppName)
+//	bucket.IndexDocument = unit.IndexDocument
+//	dag.AddResource(bucket)
+//	for _, f := range unit.Files() {
+//		object := resources.NewS3Object(bucket, filepath.Base(f.Path()), f.Path(), filepath.Join(unit.ID, f.Path()))
+//		dag.AddResource(object)
+//		dag.AddDependency(object, bucket)
+//		a.MapResourceDirectlyToConstruct(object, unit)
+//	}
+//	a.MapResourceDirectlyToConstruct(bucket, unit)
+//	return nil
+//}
 
-	bucket := resources.NewS3Bucket(unit, a.Config.AppName)
-	bucket.IndexDocument = unit.IndexDocument
-	dag.AddResource(bucket)
+func (a *AWS) expandStaticUnit(dag *core.ResourceGraph, unit *core.StaticUnit) error {
+	errs := multierr.Error{}
 	for _, f := range unit.Files() {
-		object := resources.NewS3Object(bucket, filepath.Base(f.Path()), f.Path(), filepath.Join(unit.ID, f.Path()))
-		dag.AddResource(object)
-		dag.AddDependency(object, bucket)
-		a.MapResourceDirectlyToConstruct(object, unit)
+		object, err := core.CreateResource[*resources.S3Object](dag, resources.S3ObjectCreateParams{
+			AppName:  a.Config.AppName,
+			Ref:      unit.AnnotationKey,
+			Name:     filepath.Base(f.Path()),
+			Key:      f.Path(),
+			FilePath: filepath.Join(unit.ID, f.Path()),
+		})
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		err = a.MapResourceToConstruct(object, unit)
```

Comment:
is this using the map which only allows 1 resource to 1 construct? If so i would think we would want to map the bucket and not all the objects

Reply:
Oh, yes. That's a mistake — I had meant to call `MapResourceDirectlyToConstruct`.

I'm actually pretty unclear what the difference is between `MapResourceDirectlyToConstruct` and `MapResourceToConstruct`, now that you mention it. Why do we need the second one?

Reply:
well in this scenario i would think we just map the bucket to the static unit, so we shouldnt. I think david found a way to make secrets work as well, so we shouldnt need `MapResourceDirectlyToConstruct`, its just not deleted yet but was legacy